HPCC-22828 Add "Employee Number" to user account

kunalaswani opened the pull request:
Employee Number field added to Add User, User Details Widget, User Query Widget.

Signed-off-by: Kunal Aswani <Kunal.Aswani@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Process of PR-13015, label: HPCC-22828 starts now.
The reason of this test is: Did not build yet.
Commit ID: 18b9b4ee4f51bd1d42c01f4954bf33d3ac3e3fc6
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 18b9b4ee4f51bd1d42c01f4954bf33d3ac3e3fc6
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   58 sec (00:00:58) |  253 sec (00:04:13) |    0 sec (00:00:00) |   72 sec (00:01:12) |   24 sec (00:00:24) |    0 sec (00:00:00) |   16 sec (00:00:16) |  423 sec (00:07:03) |

kunalaswani commented:
@GordonSmith  and @miguelvazq please review

miguelvazq commented:
@kunalaswani given this has translations (which some are already merged for 7.6.0) should this not target 7.6.0 or 7.8.0?

richardkchapman commented:
7.6.0 would seem preferable

kunalaswani commented:
Ok i will rebase

kunalaswani committed:
HPCC-22828 Add "Employee Number" to user account

Employee Number field added to Add User, User Details Widget, User Query Widget.

Signed-off-by: Kunal Aswani Kunal.Aswani@lexisnexisrisk.com

HPCCSmoketest commented:
Process of PR-13015, label: HPCC-22828 starts now.
The reason of this test is: New commit.
Commit ID: ffba6b27b37abcec5ee03be1fa4cee9945cb33cf
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: ffba6b27b37abcec5ee03be1fa4cee9945cb33cf
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   45 sec (00:00:45) |  250 sec (00:04:10) |    0 sec (00:00:00) |   83 sec (00:01:23) |   23 sec (00:00:23) |    0 sec (00:00:00) |   16 sec (00:00:16) |  417 sec (00:06:57) |

HPCCSmoketest commented:
Process of PR-13015, label: HPCC-22828 starts now.
The reason of this test is: Base branch changed.
Commit ID: ffba6b27b37abcec5ee03be1fa4cee9945cb33cf
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: ffba6b27b37abcec5ee03be1fa4cee9945cb33cf
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   46 sec (00:00:46) |  257 sec (00:04:17) |    0 sec (00:00:00) |   88 sec (00:01:28) |   26 sec (00:00:26) |    0 sec (00:00:00) |   16 sec (00:00:16) |  433 sec (00:07:13) |
